core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java

```diff
@@ -167,6 +166,18 @@ public void run() {
                         if (serverConfig != null) {
                             isSDKEnabled = serverConfig.source.isSourceEnabled;
                             if (isSDKEnabled) {
+                                // Initialise dataPlaneUrl based on Residency server
```

Comment:
169 to 174 should move to a different private function, something like
`dataPlaneUrl = getDataPlaneUrlWrtResidencyConfig(serverConfig, config)`